lib/segment/src/types.rs

```diff
@@ -293,11 +294,12 @@ pub const fn default_quantization_oversampling_value() -> Option<f64> {
 }
 
 /// Additional parameters of the search
-#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, Copy, PartialEq, Default)]
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Validate, Clone, Copy, PartialEq, Default)]
 #[serde(rename_all = "snake_case")]
 pub struct SearchParams {
     /// Params relevant to HNSW index
-    /// /// Size of the beam in a beam-search. Larger the value - more accurate the result, more time required for search.
+    /// Size of the beam in a beam-search. Larger the value - more accurate the result, more time required for search.
+    #[validate(range(min = 4))]
```

Comment:
Is this a valid limit? `hnsw_ef >= 4`

This is derived from [here](https://github.com/qdrant/qdrant/blob/d8f8a3666b93b9b7c29ec5a0f57869215d27a84f/lib/segment/src/types.rs#L341-L342).

Reply:
I would say no, let's not over-validate

Reply:
It should work with `hnsw_ef > 0`, who knows what for? Maybe some experiments

Reply:
Alright! I'll revert it for now and suggest these changes (with experiment) in a new PR to not delay the other changes further.